superset-frontend/src/explore/components/controls/AnnotationLayerControl/index.tsx

```diff
@@ -175,7 +214,7 @@ class AnnotationLayerControl extends React.PureComponent {
 
   render() {
     const { addedAnnotationIndex } = this.state;
-    const addedAnnotation = this.props.value[addedAnnotationIndex];
+    const addedAnnotation = this.props.value[addedAnnotationIndex!];
```

Comment:
Since `this.state.addedAnnotationIndex` possibly nullable, please avoid the bang operator.

```
const addedAnnotation = this.props.value[addedAnnotationIndex];
```